script.js

```diff
@@ -2,11 +2,164 @@
+  pageNavigation.appendChild(searchResult)
+
+  searchBar.addEventListener("keyup", searchThroughEpisodes)
+
+  function searchThroughEpisodes(e) {
+    let searchString = e.target.value.toUpperCase();
+    let allEpisodesOnPage = Array.from(getEpisodesOnPage);
+    let newEpisodes = [];
+
+    allEpisodesOnPage.forEach(episode => {
+
+      if (episode.innerText.toUpperCase().includes(searchString)) {
+        episode.style.display = "";
+        newEpisodes.push(episode);
+        searchResult.innerText = `Displaying ${newEpisodes.length}/${getEpisodesOnPage.length} episodes `;
+      } else {
+        episode.style.display = "none";
+        searchResult.innerText = `Displaying ${newEpisodes.length}/${getEpisodesOnPage.length} episodes `;
+      }
+      if (searchString.length === 0) {                                                                      //I have this condition, so when you type "winter" for example and get a result of 10/73, then when you delete it all you want the result to be 73 and not 73/73
+        searchResult.innerText = `Displaying ${getEpisodesOnPage.length} episodes `;
+      }
+    });
+
+  }
 }
 
+//level 300
+function selectEpisode() {
+  let chooseEpisode = document.createElement("label");
+  chooseEpisode.innerText = "Choose an episode";
+  chooseEpisode.classList.add("myChooseEpisode")                                                           //setAttribute can also be used to add class.
+
+  let selectAnEpisode = document.createElement("select");
+  selectAnEpisode.setAttribute("id", "episodes")
+
+  let pageNavigation = document.querySelector("nav")                                                      //I declared this variable again (there is one in level 200), did not want to have anything in the global scope, so not sure if it is best practice?
+  pageNavigation.prepend(chooseEpisode, selectAnEpisode);
+
+  let getEpisodesTitles = document.getElementsByClassName("episodeName")
+
+  for (let titleIndex = 0; titleIndex < getEpisodesTitles.length; titleIndex++) {
+    let episodeOption = document.createElement("option");
+    selectAnEpisode.appendChild(episodeOption);
+
+    episodeOption.innerText = getEpisodesTitles[titleIndex].innerText;
+  
+  }
+
+  selectAnEpisode.addEventListener("change", selected)
+  function selected(e) {
+    let selectedOption = e.target.value;
+    console.log(e)
+    let getEpisodesOnPage = document.getElementsByClassName("episodeWrapper");                             //I declared this variable again (there is one in level 200), did not want to have anything in the global scope, so not sure if it is best practice?
+    let allEpisodesOnPage = Array.from(getEpisodesOnPage);
+
+    allEpisodesOnPage.forEach( episode => {
+      if (episode.innerText.includes(selectedOption) ) {
+        episode.style.display = "";         
+      } else  {
+        episode.style.display = "none";
+      }
+    })
+
+  }
+   
+}
```

Comment:
To debug the issue you have selecting episodes, I added some `console.log` lines to check the values of your variables.
```
console.log(selectedOption) // add on line 124
console.log(episode.innerText) // add on line 129
```
I get output that looks like this
`Winter is Coming - S01E01`
`Winter is Coming -  S01E01 Lord Eddard Stark, ruler of the North.....`

Can you see the difference? There's are two spaces after the `-` in your `episode.innerText`. 

You have an extra space because of line 38
```
title.innerHTML = `${list[index].name} - <span> ${getSeasonAndEpisodeNumber(list, index)} </span>`;
```
`innerText` will strip out the `<span>` tags but keep the spacing you have. That's why you get a double space after the `-`. If I remove spaces you have within the span tag then it works.
```
title.innerHTML = `${list[index].name} - <span>${getSeasonAndEpisodeNumber(list, index)}</span>`;
```

Reply:
@louisechow thank you thank you thank you, this was the most the difficult part for me, and spent hours on it to figure out the issue, but you spotted it so quickly, amazing.